packages/jsapi-nodejs/src/loaderUtils.ts

```diff
@@ -133,7 +166,19 @@ export async function loadDhModules({
                 .replace(`export default dh;`, `module.exports = dh;`);
             }
 
-            if (serverPath === 'jsapi/dh-internal.js') {
+            if (serverPath === DH_INTERNAL_MODULE) {
+              if (error != null) {
+                // `dh-internal.js` module is being removed from future versions
+                // of DH core, but there's not a great way for this library to
+                // know whether it's present or not. Treat 404s as empty content
+                // to make things compatible with both configurations.
+                if (error instanceof HttpError && error.statusCode === 404) {
+                  return '';
```

Comment:
This should really be the only behavior change from the consumers perspective, and I don't think there's any real scenarios where this file would 404 when a server is up and running except our future scenario when we remove the file.